File: algorithm/python/BOJ10845.py
# problem : https://www.acmicpc.net/submit/10845

import sys

N = int(sys.stdin.readline().rstrip())
queue = list()
answer = list()

for i in range(0, N):
    cmd = sys.stdin.readline().rstrip()

    if cmd.startswith("push"):
        num = int(cmd[5:])
        queue.append(num)
    elif cmd.startswith("pop"):
        if len(queue) == 0:
            answer.append(-1)
        else:
            answer.append(queue[0])
            queue = queue[1:len(queue)]
    elif cmd.startswith("size"):
        answer.append(len(queue))
    elif cmd.startswith("empty"):
        if len(queue) == 0:
            answer.append(1)
        else:
            answer.append(0)
    elif cmd.startswith("front"):
        if len(queue) == 0:
            answer.append(-1)
        else:
            answer.append(queue[0])
    else:
        if len(queue) == 0:
            answer.append(-1)
        else:
            answer.append(queue[len(queue) - 1])

for item in answer:
    print(item)

File: algorithm/typescript/BOJ10845.ts
// problem : https://www.acmicpc.net/problem/10845

const fs = require("fs");

const inputData: string[] = fs.readFileSync("/dev/stdin").toString().split("\n");

const N: number = Number(inputData[0]);
const queue: number[] = new Array();
const answer: number[] = new Array();

for (let i = 1; i <= N; i++) {
  const cmd: string = inputData[i];

  if (cmd.startsWith("push")) {
    const num: number = Number(cmd.split(" ")[1]);
    queue.push(num);
  } else if (cmd.startsWith("pop")) {
    if (queue.length === 0) {
      answer.push(-1);
    } else {
      const num: number[] = queue.splice(0, 1);
      answer.push(num[0]);
    }
  } else if (cmd.startsWith("size")) {
    answer.push(queue.length);
  } else if (cmd.startsWith("empty")) {
    if (queue.length === 0) {
      answer.push(1);
    } else {
      answer.push(0);
    }
  } else if (cmd.startsWith("front")) {
    if (queue.length === 0) {
      answer.push(-1);
    } else {
      answer.push(queue[0]);
    }
  } else {
    if (queue.length === 0) {
      answer.push(-1);
    } else {
      answer.push(queue[queue.length - 1]);
    }
  }
}

console.log(answer.join("\n"));
